### PR TITLE
LightGBM in Darts

### PR DESCRIPTION
**Pull request by mkos**
Adds LGBM model to collection of RegressionModels.

### CODE REVIEW

File: requirements/core.txt [hunk 1]
```diff
@@ -11,3 +11,4 @@ scikit-learn>=0.24.0
 filterpy>=1.4.5
 joblib>=1.0.0
 requests>=2.25.0
+lightgbm==3.2.1
```
Comment (hrzn) [inline requirements/core.txt:14]: Any chance we could set it >= ?

**Review by hrzn — COMMENTED** [review]
This is very nice, just got a couple of small comments.

File: darts/tests/test_regression_models.py [hunk 2]
```diff
@@ -344,7 +350,7 @@ def test_models_denoising(self):
 
     def test_models_denoising_multi_input(self):
         # for every model, test whether it correctly denoises ts_sum_2 using ts_random_multi and ts_sum_2 as inputs
-        self.helper_test_models_accuracy(self.ts_sum2, self.ts_cov2, 19)
+        self.helper_test_models_accuracy(self.ts_sum2, self.ts_cov2, 19.5)
```
[the remark this answers is not in the document]
Comment (mkos) [reply darts/tests/test_regression_models.py:353]: Yes. I didn't investigate it though - I'm assuming that since the model has more capacity (or degrees of freedom) it's slightly overfitting and hence slightly worse job. Probably tuning parameters (the model is ran on defaults) would increase accuracy but I thought this is good enough. I spoke with Tomas and he said that he picked this threshold empirically.

File: darts/models/gradient_boosted_model.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+from ..logging import get_logger
+from typing import Union, Optional, Sequence, List, Tuple
+from .regression_model import RegressionModel
+from ..timeseries import TimeSeries
+import lightgbm as lgb
+
+logger = get_logger(__name__)
+
+
+class GradientBoostedModel(RegressionModel):
```
[the remark this answers is not in the document]
Comment (mkos) [reply darts/models/gradient_boosted_model.py:30]: No problem, even better - I called it initially that way, but I changed it after I saw that all remaining implementations are called with the algo name and not the lib it is in.

**Review by hrzn — APPROVED** [review]
LGTM (maybe @tomasvanpottelbergh can also have a quick look)
